Adds dependencies, making a PR, and updating documentation to NDG

nsryan2 opened the pull request:
This PR adds a section for all the dependencies found on the PyNE website to the New Developer's Guide, along with information of how to install them, and some relavent resources for that dependency. It also adds a section on how to make a pull request and updating documentation with the notable inclusion of how to include a notes file for the PR. #561

nsryan2 committed:
Add information about PR practices and news files from the template.

nsryan2 committed:
Adds dependencies, making a PR, and updating documentation

This PR adds a section for all the dependencies found on the PyNE website, along with information of how to install them, and some relavent resources for that dependency. It also adds a section on how to make a pull request and updating documentation with the notable inclusion of how to include a notes file for the PR.

nsryan2 committed:
Create updating_ndg_documentation

gonuke commented on docs/devsguide/new_dev_guide.rst:
```diff
@@ -209,5 +211,374 @@ text (Vim, TextEdit, Nano, etc.) and doesn’t invisibly add characters to the
 Sphinx will do everything else under the hood as described `here 
-<http://pyne.io/devsguide/website.html>`__. Finally, commit these changes to 
-your forked version and submit a pull request.
+<http://pyne.io/devsguide/website.html>`__. Before you make the pull request, 
+make sure that you include a news file named after the purpose of your pr 
```
```suggestion
<http://pyne.io/devsguide/website.html>`__. Before you make the pull request (PR), 
make sure that you include a news file named after the purpose of your PR 
```

gonuke commented on docs/devsguide/new_dev_guide.rst:
```diff
@@ -209,5 +211,374 @@ text (Vim, TextEdit, Nano, etc.) and doesn’t invisibly add characters to the
+through Anaconda will result in many of the packages and dependencies being satisfied.
+
+-------
+Website
```
Recommend putting the PyNE Library dependencies first.  We expect many more people will want to build the library without building the docs/website, so those dependencies should be more prominent.

nsryan2 replied:
I didn't think of that, but that makes a lot of sense to me too. I'll include it in the next commit.

gonuke commented on docs/devsguide/new_dev_guide.rst:
```diff
@@ -209,5 +211,374 @@ text (Vim, TextEdit, Nano, etc.) and doesn’t invisibly add characters to the
+your forked version and submit a pull request. 
+
+---------------------
+Making a Pull Request
```
I wonder if this section fits better above the section on modifying documentation?  Making a pull request is ubiquitous to contributing to PyNE and prerequisite knowledge to updating the docs.

gonuke replied:
Also, the instruction about adding a news file might fit better under this generic info on making a pull request.

nsryan2 replied:
I agree, the PR section does make more sense above the documentation section. I've moved it, and the information about the news file with my latest commit

gonuke commented on docs/devsguide/new_dev_guide.rst:
```diff
@@ -209,5 +211,374 @@ text (Vim, TextEdit, Nano, etc.) and doesn’t invisibly add characters to the
+---------------------
+The pull request should contain a description of the the pull request, any labels 
+(such as bug, docs, test), projects (such as Transition to PyMOAB), and a reference
+to the issue that led to the pr (which you can do by inserting '@issue_number' to 
```
```suggestion
to the issue that led to the PR (which you can do by inserting '#issue_number' to 
```

gonuke commented on docs/devsguide/new_dev_guide.rst:
```diff
@@ -209,5 +211,374 @@ text (Vim, TextEdit, Nano, etc.) and doesn’t invisibly add characters to the
+---------------------
+Making a Pull Request
+---------------------
+The pull request should contain a description of the the pull request, any labels 
```
```suggestion
The pull request (PR) should contain a description of the the pull request, any labels 
```

gonuke commented on docs/devsguide/new_dev_guide.rst:
```diff
@@ -209,5 +211,374 @@ text (Vim, TextEdit, Nano, etc.) and doesn’t invisibly add characters to the
+#. `Beginner’s Guide <https://wiki.python.org/moin/BeginnersGuide>`_
+
+
+**Ruby:**
```
Ruby is only required on MacOS, and only then to install homebrew.  Perhaps there should be platform-specific sections?

nsryan2 replied:
Thank you for the feedback on my pull request!
I am toying with how to include platorm-specific install instructions for these dependencies. I feel like this document is getting really long, which isn't a bad thing, but it might feel overwhelming for New Developer's. I've moved the instructions for installing ruby into the MacOS instructions for installing git, which also outline how to install homebrew through ruby. 

Do you think it would be appropriate to find links to installation instructions for each of the dependencies so that each platform is covered without adding more sections to the document? 

Some of the dependencies have this already, but I thought that Homebrew worked for other platforms so I opted to include brew install and pip install (if I'm not mistaken, pip a python tool and doesn't depend on platform) instructions for dependencies that lacked a lot of documentation (or where the documentation I was reading through didn't seem to indicate a difference for the various platforms).

gonuke replied:
I agree that it is challenging to balance being thorough with not being intimidating to new user/developers!

While brew may work on other platforms, it is typically not in the top 3 ways that people install software on those platforms.  One day someone will make the package manager to rule them all, but so far, brew is not that....

I know that tables are difficult to maintain in RST, but I wonder if some kind of table with each row being a different dependency, and each column being a different platform???

Otherwise, we might want to just point folx to different files for each platform (and you might not be responsible for all the platforms.)

gonuke replied:
I see you moved the Ruby section up to the description of installing homebrew.... do you need it here still?

katyhuff replied:
I think different files for each platform is a good plan.

nsryan2 replied:
I've created three separate pages for Mac, Linux, and Windows dependencies which have installation instructions for each platform. These pages are now linked in a section before the one on installing PyNE, and the original section for dependencies was removed.

katyhuff replied:
Hm. I'm not sure removing the original list of dependencies is a good idea.

(Also, many mac users don't use homebrew. I, for example, use macports.)

gonuke reviewed (changes requested):
Thanks @nsryan2 for the hard work of digging out all these dependencies.  It appears that the list of dependencies, and the instructions for installing them, are specific to homebrew on MacOS.  We should consider how to structure this to either be more universal, or to clearly mark it as MacOS instructions.

I noticed in reviewing this that this page is never built/displayed on our website!!  I think it's because of a filename typo in `devsguide/index.rst`.  Perhaps you can fix that, too?

nsryan2 committed:
Apply grammatical and formatting suggestions from code review

Co-authored-by: Paul Wilson <paul.wilson@wisc.edu>

nsryan2 committed:
Re-arrange sections and information

This commit moves the website dependencies below the PyNE dependencies, changes the order of the Updating Documentation and PR sections, and moves the information on the ruby installation into the MacOS specific instructions to install git, below the install homebrew subheading with an additional link on other ways to install homebrew other than through ruby.

nsryan2 commented:
I found the error that was preventing this page from showing up in the devsguide/index.html, this page was missing an anchor (I think that's what they're called) at the top. I believe that I've fixed the problem with my second commit after the suggestions on the original PR, I'm just now realizing that I forgot to put that in the description.

gonuke commented:
The anchor may fix it (??) but I think a key issue is that the links in `index.rst` refer to `new_devs_guide` but the file is called `new_dev_guide` (notice the letter "s").  I might be more transparent to just change `index.rst` to point to the correct file since having the filename and the anchor name be slightly different is going to cause someone else confusion down the line.

nsryan2 committed:
remove the s from new_devs_guide

nsryan2 committed:
remove the s from the new_devs_guide anchor

katyhuff commented on docs/devsguide/new_dev_guide.rst:
```diff
@@ -100,6 +113,8 @@ each dependency. For example, PyNE is currently supported in conda
 install from 2.7.\* - 3.6.*, and attempting to install with a default
 version of 3.7.\* will result in errors. Setting your version to be
 compatible before you begin will help ensure you are successful.
+For a comprehensive list of all the dependencies of PyNE, examine
```
```suggestion
For a comprehensive list of all the software that PyNE depends on, examine
```

katyhuff commented on docs/devsguide/new_dev_guide.rst:
```diff
@@ -129,16 +144,14 @@ create an environment for the conda install method.
 
-5. Before continuing with the PyNE `Conda Installation <http://pyne.io/install/
-conda.html#conda>`__ ensure that the Python version is correct.
+5. Before continuing with the PyNE installation ensure that the Python version is correct.
```
```suggestion
5. Before continuing with the PyNE installation, ensure that the Python version is correct.
```

katyhuff commented on docs/devsguide/new_dev_guide.rst:
```diff
@@ -201,6 +216,21 @@ the “low hanging pinoli” label to display more issues with the same tag.
+---------------------
+Making a Pull Request
+---------------------
+The pull request (PR) should contain a description of the the pull request, any labels 
```
```suggestion
The pull request (PR) should contain a description of the the changes, appropriate labels 
```

katyhuff commented on docs/devsguide/new_dev_guide.rst:
```diff
@@ -201,6 +216,21 @@ the “low hanging pinoli” label to display more issues with the same tag.
+to the issue that led to the PR, which you can do by inserting '#issue_number' to 
+the description.
+
+The more specific your description, the better you employ the labels, and 
```
```suggestion
A specific description, accurate labels, and appropriate
```

katyhuff commented on docs/devsguide/new_dev_guide.rst:
```diff
@@ -201,6 +216,21 @@ the “low hanging pinoli” label to display more issues with the same tag.
+the description.
+
+The more specific your description, the better you employ the labels, and 
+project selections the easier it will be for other contributors to review your work.
```
```suggestion
project selections will help other contributors to discover and review your work more easily.
```

katyhuff commented on docs/devsguide/new_dev_guide.rst:
```diff
@@ -201,6 +216,21 @@ the “low hanging pinoli” label to display more issues with the same tag.
+The more specific your description, the better you employ the labels, and 
+project selections the easier it will be for other contributors to review your work.
+Adding a reference to the issue the pull request will allow people to see the issue 
+inspired it, and any conversation about the issue. Before you make the pull request (PR), 
```
```suggestion
inspired it alongside any conversation about the issue. Before you make the pull request (PR), 
```

katyhuff commented on docs/devsguide/new_dev_guide.rst:
```diff
@@ -201,6 +216,21 @@ the “low hanging pinoli” label to display more issues with the same tag.
+Adding a reference to the issue the pull request will allow people to see the issue 
+inspired it, and any conversation about the issue. Before you make the pull request (PR), 
+make sure that you include a news file named after the purpose of your PR 
+following the template located in the news folder.
```
```suggestion
following the template located in the `news` directory.
```

gonuke reviewed (changes requested):
I think just one last comment - thanks @nsryan2

nsryan2 committed:
Apply suggestions from code review

Co-authored-by: Katy Huff <katyhuff@users.noreply.github.com>

nsryan2 committed:
Add files via upload

nsryan2 committed:
Update new_dev_guide.rst

gonuke commented on docs/devsguide/macosdep.rst:
```diff
@@ -0,0 +1,352 @@
+
+If you have homebrew installed on your device, you can install GCC by entering ::
+
+	$ brew install gcc
```
```suggestion
	$ brew install cmake
```

gonuke commented on docs/devsguide/linuxdep.rst:
```diff
@@ -0,0 +1,361 @@
+
+``To Install:``
+
+Anaconda is a comprehensive Python package 
```
What about native python installs using `apt-get` - do we want to support that?

gonuke commented on docs/devsguide/linuxdep.rst:
```diff
@@ -0,0 +1,361 @@
+
+``To Install:``
+
+To download GCC, visit `Installing GCC <https://gcc.gnu.org/install/index.html>`_
```
This should definitely rely on a package manager.  Either `apt-get` or `conda`, but if we send folx to install GCC from source, they may never return...

gonuke commented on docs/devsguide/linuxdep.rst:
```diff
@@ -0,0 +1,361 @@
+
+``To Install:``
+
+To download CMake, visit `Downloads <https://cmake.org/download/>`_.
```
Again, `apt-get` and/or `conda` for this

gonuke commented on docs/devsguide/linuxdep.rst:
```diff
@@ -0,0 +1,361 @@
+
+#. Install with sudo apt ::
+
+    $ sudo apt-get install python-numpy
```
If you are steering people towards Anaconda, then they should have/get `numpy` form their conda build.  The same goes for most of these python packages

gonuke commented on docs/devsguide/linuxdep.rst:
```diff
@@ -0,0 +1,361 @@
+
+BLAS (Basic Linear Algebra Subroutines) coordinates operations on vectors and matrices.
+
+``To Install:``
```
This is available from `apt-get` and maybe `conda`.

gonuke commented on docs/devsguide/linuxdep.rst:
```diff
@@ -0,0 +1,361 @@
+
+LAPACK (Liner Algebra Package) is a software library for numerical liner algebra.
+
+``To Install:``
```
Ideally, from `apt-get` or `conda`

gonuke commented on docs/devsguide/linuxdep.rst:
```diff
@@ -0,0 +1,361 @@
+
+Open your command line and enter ::
+
+    $ sudo apt-get install -y python3-sphinxcontrib.bibtex
```
You mention Python 2 support at the top.... this is python3 only.

gonuke commented on docs/devsguide/linuxdep.rst:
```diff
@@ -0,0 +1,361 @@
+
+#. Download the `source code <https://pypi.org/project/cloud_sptheme/>`_ .
+
+#. Un-zip the file (if necessary) and, in the command line, "cd" into the folder.
```
at least `pip` install this? (but that adds a new package manager...)

gonuke commented on docs/devsguide/linuxdep.rst:
```diff
@@ -0,0 +1,361 @@
+be satisfied, but it never hurts to make sure. 
+
+
+``To Install:``
```
`apt-get` or `conda`

gonuke reviewed (changes requested):
I'm afraid that we may have nudged you into the chaos of multiple package managers on multiple platforms - most certainly NOT your fault!  I have made a few comments here about the linux install, but don't know enough about the MAC install to know if similar comments are warranted.

nsryan2 committed:
fix cmake installation for MacOS

nsryan2 committed:
Add sudo apt-get install instructions for every dependency

nsryan2 committed:
add python 2 and 3 instructions where needed

katyhuff commented:
Hi @nsryan2 . Let's look over this together holistically at some point. I'm not sure I understand the structure you're going for. While you've definitely added a lot of useful information here, I'm not sure it has made the workflow for a new user much simpler. For example, the sudo apt-get commands are great, but probably should be reported as one, single sudo apt-get command, rather than one command for each package. That is, one can install packageA and packageB with `sudo apt-get install packageA packageB`. There seem to be a few places where the extra detail might lead the user to feel installation is more complex than it truly is.  

Let's discuss?

nsryan2 commented:
> Hi @nsryan2 . Let's look over this together holistically at some point. I'm not sure I understand the structure you're going for. While you've definitely added a lot of useful information here, I'm not sure it has made the workflow for a new user much simpler. For example, the sudo apt-get commands are great, but probably should be reported as one, single sudo apt-get command, rather than one command for each package. That is, one can install packageA and packageB with `sudo apt-get install packageA packageB`. There seem to be a few places where the extra detail might lead the user to feel installation is more complex than it truly is.
> 
> Let's discuss?

@katyhuff I definitely don't want new users to feel like this is more complicated than it is, do you think it would be appropriate to make this PR a draft until we have talked about it? I'll schedule a meeting with you so we can talk about how these documents can be structured and informed.

katyhuff commented:
Sounds great! I think you've done lots of great work here -- it just may need a little restructuring!

nsryan2 committed:
Add dependencies, simplify layout, add download links

This PR adds links to the source of dependencies for PyNE, the website, and optional dependencies as well. It also adds download links for a tar or zip file, and a link to github.

nsryan2 committed:
Simplify NDG and add links where appropriate

nsryan2 committed:
Add new info on updating the changelog

nsryan2 committed:
Update PR info for clarity

nsryan2 commented:
@katyhuff I've updated the source install instructions to be links, and simplified the New Developer's Guide in accordance with our conversation the other day.

nsryan2 committed:
Delete macosxdep, as it is covered in the install information

nsryan2 committed:
Delete linuxdep as it is covered in the install info

nsryan2 committed:
Delete windowsdep.rst

nsryan2 committed:
Address Imposter syndrome 

Using the source for the disclaimer that @kkiesling found on YT's github, this commit minimally adapts the source disclaimer to fit into the Guide for New Developers. It also adds a disclaimer in accordance with the license of the source material with links to the original.

nsryan2 committed:
rebase said that the changelog needed to be merged

nsryan2 committed:
resolve changelog conflicts

nsryan2 committed:
update the optional depencencies for linus_source install and osx install

opotowsky commented on CHANGELOG.rst:
```diff
@@ -13,21 +13,23 @@ v0.7.0 RC2
-   * Use new sampler for both voxel and sub-voxel R2S.
-   * Define cell_mats for subvoxel equals False
-   * Function Sampler::mesh_tag_data to be compatible for both DEFAULT and SUBVOXEL mode
+* Subvoxel R2S allows activation and photon source sampling by 
```
It looks like your editor is automatically removing spacing that is required for this to render on github.

opotowsky replied:
(The above three spaces is actually required)

opotowsky replied:
Actually is this just not rebased?

nsryan2 replied:
I rebased it earlier (it created a bunch of the non-sense above), I'll do it again though.

opotowsky commented on docs/devsguide/new_dev_guide.rst:
```diff
@@ -92,15 +35,11 @@ developers, it is recommended that you install PyNE from the source.
 Make sure that your device has all of the dependencies downloaded. You 
 can check this by typing the name of the program into your command line 
-followed by "-v" (e.g. $ python -v). If you are just starting as a 
+followed by "--version" (e.g., $ python --version). If you are starting as a
```
can you surround these with `` `x` ``  to render as code?

nsryan2 replied:
I tried it on my version of the website, and it rendered as italics. Is that how it's supposed to look?

opotowsky replied:
I think the --version in quotes would render as italicized? but $ python --version can have double ` around it to render as monospaced

opotowsky replied:
the same that single one does in this comment: `$ python --version`

opotowsky replied:
Interesting. On github, the quotes don't do anything (because it's either single * or double ** that does italics).

nsryan2 replied:
I think I figured it out, because its sphinx and the theme it requires :code: before something in `` will render in the same way as in markdown.

nsryan2 replied:
I'll rebase and include this addition in the push at the end

nsryan2 committed:
add minimum python version

nsryan2 committed:
Delete updating_ndg_documentation

nsryan2 committed:
Update readme to look like the current version

nsryan2 committed:
Merge branch '0.7.0-rc' into develop

opotowsky commented:
I'm confused why even after your rebase the changelog is still messed up! I'm leaving my computer for the evening but am happy to help with this another time

nsryan2 commented:
> I'm confused why even after your rebase the changelog is still messed up! I'm leaving my computer for the evening but am happy to help with this another time

Me too, my other rebases didn't have the same effect. Thanks for the input!

bam241 commented:
This has been taking over by #1315 
closing for now.
